wp-includes/class-wp-web-app-manifest.php

```diff
@@ -83,15 +83,51 @@ public function manifest_link_and_meta() {
 				?>
 				<meta name="apple-mobile-web-app-capable" content="yes">
 				<meta name="mobile-web-app-capable" content="yes">
+
 				<?php
 				$icons = isset( $manifest['icons'] ) ? $manifest['icons'] : array();
 				usort( $icons, array( $this, 'sort_icons_callback' ) );
 				$icon = array_shift( $icons );
-				?>
-				<?php if ( ! empty( $icon ) ) : ?>
-					<link rel="apple-touch-startup-image" href="<?php echo esc_url( $icon['src'] ); ?>">
-				<?php endif; ?>
-				<?php
+
+				$images = array();
+				if ( ! empty( $icon ) ) {
+					$images[] = array( 'href' => $icon['src'] );
+				}
+
+				/**
+				 * Filters splash screen images for Safari on iOS.
+				 *
+				 * @param array $images {
+				 *     Array of splash screen images and their attributes.
+				 *
+				 *     @type array ...$0 {
+				 *         Array of splash screen image attributes.
+				 *
+				 *         @type string $href  URL of splash screen image. Required.
+				 *         @type string $media Media query for when the splash screen image should be used.
+				 *     }
+				 * }
+				 */
+				$images = apply_filters( 'apple_touch_startup_image', $images );
+
+				foreach ( $images as $key => $image ) {
+					if ( ! is_array( $image ) ) {
+						continue;
+					}
+
+					$url = isset( $image['href'] ) ? esc_url( $image['href'], array( 'http', 'https' ) ) : '';
```

Comment:
```suggestion
					$url = isset( $image['href'] ) ? esc_url_raw( $image['href'], array( 'http', 'https' ) ) : '';
```

Reply:
I think you're suggesting `esc_url_raw()` here because you suggest `esc_attr( $url )` further down but I'm not sure this treatment is equal to `esc_url()`.

Reply:
Humm, yeah. I guess `esc_url()` is not the same as `esc_url_raw()`+`esc_attr()` after looking at the source. So nevermind.